runtime/gc_glue_java/MarkingDelegate.cpp

```diff
@@ -294,27 +313,13 @@ MM_MarkingDelegate::scanContinuationNativeSlots(MM_EnvironmentBase *env, omrobje
 	const bool isGlobalGC = true;
 	const bool beingMounted = false;
 	if (MM_GCExtensions::needScanStacksForContinuationObject(currentThread, objectPtr, isConcurrentGC, isGlobalGC, beingMounted)) {
-		StackIteratorData4MarkingDelegate localData;
-		localData.markingDelegate = this;
-		localData.env = env;
-		localData.fromObject = objectPtr;
-
 		bool stackFrameClassWalkNeeded = false;
 #if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
 		stackFrameClassWalkNeeded = isDynamicClassUnloadingEnabled();
 #endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
 
-		GC_VMThreadStackSlotIterator::scanContinuationSlots(currentThread, objectPtr, (void *)&localData, stackSlotIteratorForMarkingDelegate, stackFrameClassWalkNeeded, false);
-
-#if JAVA_SPEC_VERSION >= 24
 		J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, objectPtr);
-		GC_ContinuationSlotIterator continuationSlotIterator(currentThread, continuation);
-
-		while (J9Object **slotPtr = continuationSlotIterator.nextSlot()) {
-			doContinuationSlot(env, objectPtr, slotPtr, &continuationSlotIterator);
-		}
-#endif /* JAVA_SPEC_VERSION >= 24 */
-
+		scanContinuationNativeSlotsNoSync(env, NULL, continuation, stackFrameClassWalkNeeded);
```

Comment:
For easier undertanding what we pass as NULL you can do:
`const J9VMThread *walkThread = NULL;`
We already do the same thing a bit earlier for some other parameters when passing to needScanStacksForContinuationObject()

Reply:
`J9VMThread * const walkThread = NULL;` is correct

Reply:
change `const J9VMThread *walkThread = NULL;` to `J9VMThread *walkThread = NULL;` due to compiler error 

```
/home/linhu/openj9-openjdk-jdk24/openj9/runtime/gc_glue_java/MarkingDelegate.cpp: In member function ‘void MM_MarkingDelegate::scanContinuationNativeSlots(MM_EnvironmentBase*, omrobjectptr_t)’:
/home/linhu/openj9-openjdk-jdk24/openj9/runtime/gc_glue_java/MarkingDelegate.cpp:320:56: error: invalid conversion from ‘const J9VMThread*’ to ‘J9VMThread*’ [-fpermissive]
  320 |                 scanContinuationNativeSlotsNoSync(env, walkThread, continuation, stackFrameClassWalkNeeded);
      |                                                        ^~~~~~~~~~
 
```

Reply:
`J9VMThread *walkThread = NULL` is ok, but 'J9VMThread * const walkThread = NULL` is even stronger. We are just saying the compiler that walkThread is a const, and we will not change it (and we indeed do not, we just use it soon after to pass as a const argument). 

`const J9VMThread *walkThread = NULL` was indeed incorrect, since it was requesting that the user will not ever change the value of *walkThread, and indeed scanContinuationNativeSlotsNoSync never promised it (it's parameter is just plain `J9VMThread *`